packages/ses/src/transforms.js

```diff
@@ -76,26 +130,67 @@ export function rejectImportExpressions(src) {
   );
 }
 
-// The shim cannot correctly emulate a direct eval as explained at
-// https://github.com/Agoric/realms-shim/issues/12
-// Without rejecting apparent direct eval syntax, we would
-// accidentally evaluate these with an emulation of indirect eval. To
-// prevent future compatibility problems, in shifting from use of the
-// shim to genuine platform support for the proposal, we should
-// instead statically reject code that seems to contain a direct eval
-// expression.
-//
-// As with the dynamic import expression, to avoid a full parse, we do
-// this approximately with a regexp, that will also reject strings
-// that appear safely in comments or strings. Unlike dynamic import,
-// if we miss some, this only creates future compat problems, not
-// security problems. Thus, we are only trying to catch innocent
-// occurrences, not malicious one. In particular, `(eval)(...)` is
-// direct eval syntax that would not be caught by the following regexp.
-
-const someDirectEvalPattern = new RegExp('\\beval\\s*(?:\\(|/[/*])');
-
-// Exported for unit tests.
+/**
+ * An optional transform to place ahead of `rejectImportExpressions` to evade
+ * *that* rejection. However, it may change the meaning of the program.
+ *
+ * This evasion replaces each suspicious `import` identifier with `__import__`.
+ * If the alleged import expression appears in a JavaScript comment, this
+ * evasion will not change the meaning of the program. If it appears in a
+ * literal (string literal, regexp literal, or a template literal), then this
+ * evasion will change the contents of that literal. If it appears as code
+ * where it would be parsed as an expression, then it might or might not change
+ * the meaning of the program, depending on the binding, if any, of the lexical
+ * variable `__import__`.
+ *
+ * Finally, if the original appears in code where it is not parsed as an
+ * expression, for example `foo.import(path)`, then this evasion would rewrite
+ * to `foo.__import__(path)` which has a surprisingly different meaning.
```

Comment:
heres an example of code in the wild that would end up with a "surprisingly different meaning."
https://npmfs.com/package/@babel/helper-module-imports/7.10.4/lib/import-injector.js#L111

Reply:
interestingly, in our case the above package works because the transform also modifies the method definition. so it successfully calls `builder.__import__()`.